lib_dfu/src/main/java/no/nordicsemi/android/dfu/BaseDfuImpl.java

```diff
@@ -751,8 +751,12 @@ byte[] readNotificationResponse()
 	void restartService(@NonNull final Intent intent, final boolean scanForBootloader) {
 		String newAddress = null;
 		if (scanForBootloader) {
-			mService.sendLogBroadcast(DfuBaseService.LOG_LEVEL_VERBOSE, "Scanning for the DFU Bootloader...");
-			newAddress = BootloaderScannerFactory.getScanner().searchFor(mGatt.getDevice().getAddress());
+			final long delay = intent.getIntExtra(DfuBaseService.EXTRA_SCAN_DELAY, 0);
+			final long timeout = intent.getIntExtra(DfuBaseService.EXTRA_SCAN_TIMEOUT, 5000);
```

Comment:
The next is not a complaint, but only an observation. The following logs have been displayed while testing this PR.

```
2022-08-05 18:51:24.041 18543-22190/com.app I/DfuImpl: Restarting to bootloader mode
2022-08-05 18:51:24.042 18543-22190/com.app W/Bundle: Key no.nordicsemi.android.dfu.extra.EXTRA_SCAN_DELAY expected Integer but value was a java.lang.Long.  The default value 0 was returned.
2022-08-05 18:51:24.043 18543-22190/com.app W/Bundle: Attempt to cast generated internal exception:
    java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.Integer
        at android.os.BaseBundle.getInt(BaseBundle.java:1041)
        at android.content.Intent.getIntExtra(Intent.java:7847)
        at no.nordicsemi.android.dfu.BaseDfuImpl.restartService(BaseDfuImpl.java:754)
        at no.nordicsemi.android.dfu.BaseButtonlessDfuImpl.finalize(BaseButtonlessDfuImpl.java:112)
        at no.nordicsemi.android.dfu.ButtonlessDfuImpl.performDfu(ButtonlessDfuImpl.java:160)
        at no.nordicsemi.android.dfu.ButtonlessDfuWithoutBondSharingImpl.performDfu(ButtonlessDfuWithoutBondSharingImpl.java:106)
        at no.nordicsemi.android.dfu.DfuBaseService.onHandleIntent(DfuBaseService.java:1401)
        at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:78)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.os.HandlerThread.run(HandlerThread.java:67)
2022-08-05 18:51:24.043 18543-22190/com.app W/Bundle: Key no.nordicsemi.android.dfu.extra.EXTRA_SCAN_TIMEOUT expected Integer but value was a java.lang.Long.  The default value 5000 was returned.
2022-08-05 18:51:24.044 18543-22190/com.app W/Bundle: Attempt to cast generated internal exception:
    java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.Integer
        at android.os.BaseBundle.getInt(BaseBundle.java:1041)
        at android.content.Intent.getIntExtra(Intent.java:7847)
        at no.nordicsemi.android.dfu.BaseDfuImpl.restartService(BaseDfuImpl.java:755)
        at no.nordicsemi.android.dfu.BaseButtonlessDfuImpl.finalize(BaseButtonlessDfuImpl.java:112)
        at no.nordicsemi.android.dfu.ButtonlessDfuImpl.performDfu(ButtonlessDfuImpl.java:160)
        at no.nordicsemi.android.dfu.ButtonlessDfuWithoutBondSharingImpl.performDfu(ButtonlessDfuWithoutBondSharingImpl.java:106)
        at no.nordicsemi.android.dfu.DfuBaseService.onHandleIntent(DfuBaseService.java:1401)
        at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:78)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.os.HandlerThread.run(HandlerThread.java:67)
```

This can be easily fixed by replacing `getIntExtra` with `getLongExtra` for **delay** and **timeout**.

Reply:
Thanks, I missed that. Will fix in Monday.

Reply:
Fixed!